tests/functional/consumer/test_consumer.py

```diff
@@ -150,19 +151,24 @@ def test_consumer_killed_with_sigint(database, monkeypatch, capfd):
     process = subprocess.Popen(["orion", "hunt", "-n", EXP_NAME,
                                 "--pool-size=3", "--max-trials=6",
                                 "./success_box.py", "-x~norm(34, 3)"])
-    time.sleep(2.2)
-    process.send_signal(signal.SIGINT)
-
-    rcode = process.wait()
-    assert rcode == 1
-    capfd.readouterr()  # Suppress fd level 1 & 2
 
+    time.sleep(3)
     exp = list(database.experiments.find({'name': EXP_NAME}))
     assert len(exp) == 1
     exp = exp[0]
     assert '_id' in exp
     exp_id = exp['_id']
 
+    cc = 0
```

Comment:
This is why I would prefer to avoid the use of `Popen` in unit-tests. We could mock the methods to automatically trigger signals at the right time rather than fine-tuning sleep intervals or listening to the database to detect an event.

Reply:
But then again, who would catch the signals? I believe we need 2 processes for these kinds of tests. How would it be made synchronous, I don't know.

Reply:
python threading could do the job I guess. The advantage is that we have access to code run inside the thread so that we can mock objects inside of it. Popen totally isolate the executed process and we can't mock anything inside of it.